apps/site/src/components/pages/hub/block-data-container.tsx

```diff
@@ -53,143 +70,218 @@ export const BlockDataContainer: FunctionComponent<BlockDataContainerProps> = ({
-          ...blockVariant.properties,
-        };
+        setEntity((prevEntity) => {
+          if (prevEntity && parsedProperties) {
+            if (stringifyProperties(prevEntity.properties) === propertiesText) {
+              return prevEntity;
+            } else {
+              return {
+                ...prevEntity,
+                properties: parsedProperties,
+              };
+            }
+          }
 
-        setText(JSON.stringify(nextText, undefined, 2));
+          return null;
+        });
         previousBlockVariantsTab.current = blockVariantsTab;
       } catch (err) {
         setAlertSnackBarOpen(true);
         setBlockVariantsTab(previousBlockVariantsTab.current);
       }
     }
+  }, [blockVariantsTab, metadata, propertiesText]);
 
-    return () => {
-      const previousBlockVariant: BlockVariant | undefined =
-        metadata?.variants?.[blockVariantsTab];
+  // If the entity is changed, update the text input of the properties
+  useEffect(() => {
+    if (entity) {
+      setPropertiesText(stringifyProperties(entity.properties));
+    }
+  }, [entity, setPropertiesText]);
+
+  // If the text of the properties is updated, try and update the entity
+  useEffect(() => {
+    if (propertiesText) {
+      try {
+        const parsedProperties = JSON.parse(propertiesText);
 
-      if (previousBlockVariant) {
-        propertiesToRemove.current = Object.keys({
-          ...previousBlockVariant.properties,
-          ...previousBlockVariant.examples?.[0],
+        setEntity((prevEntity) => {
+          if (prevEntity) {
+            if (stringifyProperties(prevEntity.properties) === propertiesText) {
+              return prevEntity;
+            } else {
+              return {
+                ...prevEntity,
+                properties: parsedProperties,
+              };
+            }
+          }
+
+          return null;
         });
+      } catch (err) {
+        /* @todo - what's the correct way to handle errors here */
+        setAlertSnackBarOpen(true);
       }
-    };
-  }, [blockVariantsTab, metadata?.examples, metadata?.variants, text]);
+    }
+  }, [propertiesText, entity]);
 
   /** used to recompute props and errors on dep changes (caching has no benefit here) */
   const [props, errors] = useMemo<
-    [{ blockEntity: Entity; readonly: boolean } | undefined, string[]]
+    [
+      { blockEntity: Entity | EntityTemporal; readonly: boolean } | undefined,
+      string[],
+    ]
   >(() => {
-    const intervalForAllTime =
-      (): EntityTemporalVersioningMetadata[keyof EntityTemporalVersioningMetadata] => {
-        return {
-          start: {
-            kind: "inclusive",
-            limit: new Date(0).toISOString(),
-          },
-          end: {
-            kind: "unbounded",
-          },
-        } as const;
-      };
+    if (!entity) {
+      /* @todo - we need should handle this case and display a loading skeleton or error message */
+      return [undefined, []];
```

Comment:
This just ends up with an error in the console where something tries to access `blockEntity` but it doesn't exist. I'm not sure fixing this is within the scope of this PR unless I get some assistance